packages/lexicon/src/types.ts

```diff
@@ -287,6 +288,14 @@ export function hasProp<K extends PropertyKey>(
   return prop in data
 }
 
+export const discriminatedObject = z.object({ $type: z.string() })
```

Comment:
If I am reading this right, I think we want this type to also allow unknown keys.

E.g. right now the type is:
```ts
type DiscriminatedObject = {
    $type: string;
}
```

And I think we want:
```ts
type DiscriminatedObject = {
    $type: string;
    [x: string]: unknown;
}
```

If that's right, that could by done by tacking on a `.catchall(z.unknown())`.

Reply:
Oh -- you're right, but not for there; it's an issue in the codegen. I'll fix that.